ethcore/src/client/client.rs

```diff
@@ -520,6 +523,36 @@ impl Importer {
 
 		let mut batch = DBTransaction::new();
 
+		let ancestry_actions = self.engine.ancestry_actions(block.block(), chain.ancestry_header_iter(*parent));
+
+		let metadata = block.block().metadata().map(Into::into);
+		let is_finalized = block.block().is_finalized();
+
+		let new = ExtendedHeader {
+			header: header.clone(),
+			is_finalized: is_finalized,
+			metadata: metadata,
+			parent_total_difficulty: chain.block_details(&parent).unwrap_or_else(|| panic!("Invalid parent hash: {:?}", parent)).total_difficulty
```

Comment:
for this and other panics in the PR, please follow the codebase guideline: https://github.com/paritytech/parity/issues/1026